Applications/Spire/Include/Spire/Ui/IntegerInputWidget.hpp

```diff
@@ -1,29 +1,36 @@
 #ifndef SPIRE_INTEGER_INPUT_WIDGET_HPP
 #define SPIRE_INTEGER_INPUT_WIDGET_HPP
-#include <QSpinBox>
+#include "Spire/Ui/DecimalInputWidget.hpp"
 
 namespace Spire {
 
   //! Represents a widget for inputting whole numbers.
-  class IntegerInputWidget : public QSpinBox {
+  class IntegerInputWidget : public DecimalInputWidget {
```

Comment:
I still have hesitations that this inheritance is the right choice, but you can leave it for now.